core/src/main/java/io/questdb/cairo/CairoEngine.java

```diff
@@ -310,6 +318,108 @@ private void rename0(Path path, CharSequence tableName, Path otherPath, CharSequ
         }
     }
 
+    public final TelemetryWriterJob startTelemetry() {
+        this.telemetryWriterJob = new TelemetryWriterJob(configuration);
+
+        return this.telemetryWriterJob;
+    }
+
+    public final void storeTelemetry(short event) {
+        final MicrosecondClock clock = configuration.getMicrosecondClock();
+        final long cursor = telemetryPubSeq.next();
+        TelemetryRow row = telemetryQueue.get(cursor);
+
+        row.ts = clock.getTicks();
+        row.event = event;
+        telemetryPubSeq.done(cursor);
+    }
+
+    private class TelemetryWriterJob extends SynchronizedJob implements Closeable {
+        private final NanosecondClock nanosecondClock = configuration.getNanosecondClock();
+        private final MicrosecondClock microsecondClock = configuration.getMicrosecondClock();
+        private final CharSequence telemetryTableName = "telemetry";
+        private final QueueConsumer<TelemetryRow> myConsumer = this::toTelemetryTable;
+        private final CharSequence id;
+        private final TableWriter writer;
+
+        public TelemetryWriterJob(CairoConfiguration configuration) {
+            final CharSequence root = configuration.getRoot();
+
+            try(Path path = new Path()) {
+                if (getStatus(AllowAllCairoSecurityContext.INSTANCE, path, telemetryTableName) == TableUtils.TABLE_DOES_NOT_EXIST) {
+                    final TelemetryTableModel telemetry = new TelemetryTableModel(telemetryTableName);
```

Comment:
i should have mentioned this earlier, sorry, but table can be created via SqlCompiler and SQL, way simpler. We can pass the call site via SqlExecution context, which is everywhere CairoEngine is.

At this juncture I would suggest we create another table where we store guid
the guid should be of Long256 type

Reply:
done